src/Sylius/Behat/Context/Application/Admin/ManagingCatalogPromotionsContext.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Application\Admin;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Component\Promotion\Message\CatalogPromotionCreated;
+use Symfony\Component\Messenger\TraceableMessageBus;
+use Webmozart\Assert\Assert;
+
+final class ManagingCatalogPromotionsContext implements Context
+{
+    private ApiClientInterface $client;
+    private TraceableMessageBus $messageBus;
+
+    public function __construct(ApiClientInterface $client, TraceableMessageBus $messageBus)
+    {
+        $this->client = $client;
+        $this->messageBus = $messageBus;
+    }
+
+    /**
+     * @When I create a new catalog promotion with :code code and :name name
+     */
+    public function iCreateANewCatalogPromotionWithCodeAndName(string $code, string $name): void
+    {
+        $this->client->buildCreateRequest();
+        $this->client->addRequestData('code', $code);
+        $this->client->addRequestData('name', $name);
+        $this->client->create();
+    }
+
+    /**
+     * @Then there should be :amount new catalog promotion on the list
+     */
+    public function thereShouldBeNewCatalogPromotionOnTheList(int $amount): void
+    {
+        Assert::count($this->messageBus->getDispatchedMessages(), $amount);
+    }
+
+    /**
+     * @Then it should have :code code and :name name
+     */
+    public function itShouldHaveCodeAndName(string $code, string $name): void
+    {
+        /** @var CatalogPromotionCreated $actualMessage */
+        $actualMessage = $this->messageBus->getDispatchedMessages()[0];
+        Assert::eq($actualMessage->code, $code);
+        Assert::eq($actualMessage->name, $name);
```

Comment:
As far as I see, we already have a `get` endpoint for Catalog Promotions [here](https://github.com/Sylius/Sylius/pull/12986/files#diff-664b75ee2acc7eb49590a963ef9b3beafe2cc3d8410ebb154182e152fae76f82R24), so we should probably be able to check this data via API?

Reply:
Yup, check `Api/Admin/ManagingCatalogPromotionsContext` this is just a quick idea of how to check that the message has been dispatched to unlock other tasks but it could be moved to PHPUnit as well.

Reply:
I wonder, maybe we should have a step `And this catalog promotion should be usable` (or sth similar) in which we would check the message dispatching? Because having a CP with code and name is not related to this message at all (as it's not created by consuming this event). But for sure, checking is the message dispatched gives us a huge value - ensuring, that it can be handled by some (yet) mysterious _catalog promotions system_ 🪨

Reply:
One option would be to check even on the application level, while API response in API context. The problem with it, is how would we like to create catalog promotion on an application level

Reply:
Currently this application context is not _de facto_ application context, as it uses the API client to throw API calls 💃 Maybe we should stand with only `@api` context and there have a step for checking the created CP data (API index) and _readiness for applications (check if event is thrown)

Reply:
The last problem with checking it in two steps for me is that checking for event and API call mixes our perspectives. I would prefer to go with a separate context. But that may be also a case for refactor

Reply:
Then we either have a redundant context (application) that does not provide us with any real value, or the API context with one step that is not strictly API-related. In my opinion, the second option is less harmful and easier/faster to refactor in the future. Nevertheless - this event-related discussion is not valid in this PR any more 😄